### PR TITLE
Feature/cdnc 2263 Add toggle which can block domain failovers

### PR DESCRIPTION
**Pull request by allenchen2244**
**What Changed?**
Added support for locking down a cluster when we don't want to allow domains to be failed over to it if the cluster is already at capacity

**Why?**
In certain cases where clusters are already running at capacity failing over additional traffic can have harmful effects.

**How did you test it?**
Unit tests for both a successful failover when cluster isn't in lockdown and unit tests for failed when cluster is in lockdown.
Was also able to spin up 2 clusters using docker ui voodoo to verify end to end on local machine.

**Potential Risks**
Nothing should happen initially since default value is false and the blocking code will never be reached.

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -4387,6 +4393,13 @@ func checkPermission(
 	return nil
 }
 
+func checkFailOverPermission(config *Config, domainName string) error {
+	if config.Lockdown(domainName) {
```
Comment (Shaddoll) [inline service/frontend/workflowHandler.go:4397]: Should we allow super users to perform failover when there is a lockdown?

File: service/frontend/workflowHandler.go [hunk 2]
```diff
@@ -133,6 +133,7 @@ var (
 	errClusterNameNotSet                          = &types.BadRequestError{Message: "Cluster name is not set."}
 	errEmptyReplicationInfo                       = &types.BadRequestError{Message: "Replication task info is not set."}
 	errEmptyQueueType                             = &types.BadRequestError{Message: "Queue type is not set."}
+	errDomainInLockdown                           = &types.BadRequestError{Message: "Domain is not accepting fail overs at this time."}
```
Comment (Shaddoll) [inline service/frontend/workflowHandler.go:136]: "Domain is not accepting fail overs due to lockdown." I think it's better to tell users there is a lockdown.

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (forshakuras) [reply service/frontend/workflowHandler.go:4397]: We don't plan on making this auth based. It will purely be a flipr config

Comment (allenchen2244) [reply service/frontend/workflowHandler.go:4397]: We talked about this yesterday. The only plan right now is to have this as a flipr config not any sort of auth based solution

**Review by Shaddoll — APPROVED** [review]

**Review by davidporter-id-au — COMMENTED** [review]
Nice!

**Pull request merged**